### PR TITLE
Fixed mixed precision

### PR DESCRIPTION
**Pull request by lars-nieradzik**
binary_cross_entropy_with_logits is required for AMP. Refer to https://github.com/WongKinYiu/yolov7/blob/a207844b1ce82d204ab36d87d496728d3d2348e7/utils/loss.py#L731 in the original YOLOv7 code.

### CODE REVIEW

**Comment by Chris-hughes10** [comment]
Looks great, one minor nit, would you add a comment above to highlight that we are doing this due to mixed precision support?

**Comment by lars-nieradzik** [comment]
Okay, done.
